.github/workflows/boulder-ci.yml

```
@@ -10,9 +10,6 @@ on:
       - main
       - release-branch-*
   pull_request:
```

Per docs here: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet

I think you want:

`*`: To match branches with zero or more characters, but does not match the `/` character.
```suggestion
  pull_request:
    branches:
      - '*'
```

or

`**`: To match branches with ​zero or more characters.
```suggestion
  pull_request:
    branches:
      - '**'
```